lib/smart_proxy_pulp_plugin/pulp_api.rb

```diff
@@ -17,5 +18,11 @@ class Api < Sinatra::Base
         log_halt 503, "Pulp server '#{URI.parse(::PulpProxy::Plugin.settings.pulp_url.to_s).host}' is unknown"
       end
     end
+
+    get "/status/diskfree" do
+      pulp_disk = DiskUsage.new
+      log_halt 500, "#{PulpProxy::Plugin.settings.pulp_dir} does not exist!" unless pulp_disk.has_df_h?
```

Comment:
this probably needs to go into the class,  as well as things like:
* permissions
* no such directory
* no df binary
* df output failed (which might report all of the above)